athf/metrics/__init__.py

```
@@ -0,0 +1,338 @@
+
+def register_context_provider(
+    provider: Callable[[], tuple[Optional[str], Optional[str]]],
+) -> None:
+    """Register a callable that returns ``(hunt_id, session_id)`` for the active session.
+
+    Plugins (vault extensions) call this once at import time to plug in their
+    own session manager. ATHF core ships no provider, so without one
+    registered, ``_resolve_active_context`` returns ``(None, None)`` and
+    callers must pass ``hunt_id`` / ``session_id`` explicitly.
+    """
+    global _context_provider
```

Ruff (0.15.18)

[warning] Using the global statement to update `_context_provider` is discouraged

(PLW0603)